test/client/js/helpers/roundBadgeNumberTest.ts

```diff
@@ -6,10 +6,15 @@ describe("roundBadgeNumber helper", function () {
 		expect(roundBadgeNumber(123)).to.equal("123");
 	});
 
-	it("should return numbers above 999 in thousands", function () {
+	it("should return numbers between 1000 and 999999 with a 'k' suffix", function () {
 		expect(roundBadgeNumber(1000)).to.be.equal("1.0k");
 	});
 
+	it("should return numbers above 999999 with a 'M' suffix", function () {
+		expect(roundBadgeNumber(1000000)).to.be.equal("1.0M");
+		expect(roundBadgeNumber(1234567)).to.be.equal("1.2M");
```

Comment:
Please add a test for much higher magnitude number (with current code it would validate that it `magnitudeIndex` works` as expected.